plugin/session_buffer.py

```diff
@@ -405,5 +423,112 @@ def _present_diagnostics_async(
         for sv in self.session_views:
             sv.present_diagnostics_async()
 
+    # --- textDocument/semanticTokens ----------------------------------------------------------------------------------
+
+    def do_semantic_tokens_async(self, view: sublime.View) -> None:
+        # time.sleep(0.2)  # for testing with clangd
+        if not userprefs().semantic_highlighting:
+            return
+        if not self.session.has_capability("semanticTokensProvider"):
+            return
+        if self.session.get_capability('semanticTokensProvider.legend.tokenTypes') is None:
+            return
+        if self.session.get_capability('semanticTokensProvider.legend.tokenModifiers') is None:
+            return
+        # semantic highlighting requires a special rule in the color scheme for the View.add_regions workaround
+        if "background" not in view.style_for_scope("meta.semantic-token"):
+            return
+        self.semantic_tokens.view_change_count = view.change_count()
+        params = {"textDocument": text_document_identifier(view)}  # type: Dict[str, Any]
+        if self.semantic_tokens.result_id and self.session.has_capability("semanticTokensProvider.full.delta"):
+            params["previousResultId"] = self.semantic_tokens.result_id
+            request = Request.semanticTokensFullDelta(params, view)
+            self.session.send_request_async(request, self._on_semantic_tokens_delta)
+        elif self.session.has_capability("semanticTokensProvider.full"):
+            request = Request.semanticTokensFull(params, view)
+            self.session.send_request_async(request, self._on_semantic_tokens)
+        elif self.session.has_capability("semanticTokensProvider.range"):
+            params["range"] = region_to_range(view, view.visible_region()).to_lsp()
+            request = Request.semanticTokensRange(params, view)
+            self.session.send_request_async(request, self._on_semantic_tokens)
+
+    def _on_semantic_tokens(self, response: Optional[Dict]) -> None:
```

Comment:
Can you add the `_async` suffix? It's not consistent right now. Some methods have it and some not.